packages/hint-manifest-icons/src/hint.ts

```diff
@@ -0,0 +1,220 @@
+                const message = `Icon could not be fetched (request failed).`;
+
+                context.report(iconPath, message);
+
+                return null;
+            }
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                const message = `Icon could not be fetched (status code: ${response.statusCode}).`;
+
+                context.report(iconPath, message);
+
+                return null;
+            }
+
+            return response.body.rawContent;
+        };
+        /**
+         * Passes only for the PNG files
+         * @param iconType type specified in the manifest file
+         * @param rawContent raw datastream
+         * @param iconPath icon resource path
+         */
+        const validateImageType = (iconType: string | undefined, rawContent: Buffer, iconPath: string) => {
+            const allowedTypes = ['png', 'jpg'];
+
+            if (iconType === undefined) {
+                const message = `Icon type was not specifed`;
+
+                context.report(iconPath, message);
+
+                return false;
+            }
+
+            const specifiedType = iconType.split('/')[1];
+
+            /** Handling for the corrupt rawContent */
+            if (rawContent !== null) {
+                const image: ImageTypeResult | null = imageType(rawContent);
+
+                if (image !== null) {
+                    const isValidType: boolean = allowedTypes.some((item) => {
+                        return item === image.ext;
+                    });
+
+                    if (specifiedType !== image.ext) {
+                        const message = `Real image type (${image.ext}) do not match with specified type (${specifiedType})`;
+
+                        context.report(iconPath, message);
+                    }
+
+                    if (isValidType) {
+                        return true;
+                    }
+                }
+
+                const message = `Icon should be a valid image type ${JSON.stringify(allowedTypes)}`;
+
+                context.report(iconPath, message);
+
+                return false;
+            }
+
+            return false;
+        };
+
+        /**
+         *
+         * @param iconSizes Sizes specified in the manifest file
+         * @param iconRawData
+         * @param iconPath full path to the icon file
+         */
+        const validateSizes = (iconSizes: string | undefined, iconRawData: Buffer, iconPath: string) => {
+            if (iconSizes === undefined) {
+                context.report(iconPath, `Sizes not specifed for icon`);
+
+                return false;
+            }
+            /**
+             * sizes can be the string 'any' OR
+             * two non-negative integers without leading 0s and separated by 'x' like 144x144
+             */
+            if (iconSizes === 'any') {
+                return false;
+            }
+            const specifiedSize = iconSizes.split('x');
```

Comment:
I'd recommend a newline after the `if` (actually thought our linter required that...)
```suggestion

            const specifiedSize = iconSizes.split('x');
```

Reply:
Added line after if statements.